### PR TITLE
feat: Added Cancel button in ProgressDialog in FeedFragment

### LINKED ISSUE #2076: Allow users to cancel the LiveFeed download
**Actual Behaviour**

There is no option to cancel the LiveFeed download once the user open FeedFragment.

**Expected Behaviour**

There should also be an option to cancel the download, if the user wants.

**Steps to reproduce it**

Open FeedFragment

**Would you like to work on the issue?**

Yes

**Proposed changes**
In the progress dialog, add a cancel button to cancel loading and go back.

### PR DESCRIPTION
**Pull request by dr0pdb**
Fixes #2076 
Changes: Added Cancel button in ProgressDialog in FeedFragment

Screenshots for the change: NA

@iamareebjamal Please review.

### CODE REVIEW

**Comment by S2606** [comment]
@srv-twry Can you please provide gif for this??

File: android/app/src/main/java/org/fossasia/openevent/fragments/FeedFragment.java [hunk 1]
```diff
@@ -75,28 +78,29 @@ private void downloadFeed() {
...
-                            .getString(R.string.refresh_failed), Snackbar.LENGTH_LONG)
-                            .setAction(R.string.retry_download, view -> refresh()).show();
-                    Timber.d("Refresh not done");
-                    showProgressBar(false);
-                }, () -> {
-                    Views.setSwipeRefreshLayout(swipeRefreshLayout, false);
-                    Timber.d("Refresh done");
-                    showProgressBar(false);
-                });
+        feedLoaderDisposable =
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/fragments/FeedFragment.java]: No need to add an indent and a new line, just assign in the first statement

File: android/app/src/main/java/org/fossasia/openevent/fragments/FeedFragment.java [hunk 2]
```diff
@@ -161,6 +165,17 @@ private void setupProgressBar() {
...
         String shownMessage = String.format(getString(R.string.downloading_format), getString(R.string.menu_feed));
         downloadProgressDialog.setMessage(shownMessage);
+        downloadProgressDialog.setButton(DialogInterface.BUTTON_NEGATIVE, getString(R.string.cancel), (dialogInterface, i) -> {
+            downloadProgressDialog.dismiss();
+            disposeRxSubscriptions();
+            getActivity().onBackPressed();
+        });
+    }
+
+    void disposeRxSubscriptions() {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/fragments/FeedFragment.java]: Also check if it is already disposed

Comment (iamareebjamal) [reply android/app/src/main/java/org/fossasia/openevent/fragments/FeedFragment.java]: Also make it private

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Comment by dr0pdb** [comment]
Changes: Added proper indentation, disposable already disposed check and made method private.
GIF: 
videotogif_2018 01 10_00 27 29

@iamareebjamal Please review.

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**